src/rules/spaceInParensRule.ts

```diff
@@ -0,0 +1,222 @@
+      This rule will enforce consistency of spacing directly inside of parentheses,
+      by disallowing or requiring one or more spaces to the right of ( and to the
+      left of ). In either case, () will still be allowed. 
+      `,
+    optionsDescription: Lint.Utils.dedent`
+      There are two options for this rule:
+
+      - \`"never"\` (default) enforces zero spaces inside of parentheses
+      - \`"always"\` enforces a space inside of parentheses
+
+      Depending on your coding conventions, you can choose either option by specifying
+      it in your configuration.
+      `,
+    options: {
+      type: 'array',
+      items: [
+        {
+          enum: ['always', 'never']
+        },
+        {
+          type: 'object',
+          properties: {
+            exceptions: {
+              type: 'array',
+              items: [
+                {
+                  enum: ['{}', '[]', '()', 'empty']
+                }
+              ],
+              uniqueItems: true
+            }
+          },
+          additionalProperties: false
+        }
+      ],
+      minItems: 0,
+      maxItems: 2
+    },
+    optionExamples: [
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, "always"]
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, "never"]
+        `,
+      Lint.Utils.dedent`
+        "${RULE_NAME}": [true, "always", { "exceptions": [ "{}", "[]", "()", "empty" ] }]
+        `
+    ],
+    typescriptOnly: false,
+    type: 'style'
+  };
+
+  public static MISSING_SPACE_MESSAGE = 'there must be a space inside this paren.';
+  public static REJECTED_SPACE_MESSAGE = 'there should be no spaces inside this paren.';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const walker = new SpaceInParensWalker(sourceFile, this.getOptions());
+    return this.applyWithWalker(walker);
+  }
+}
+
+class SpaceInParensWalker extends Lint.RuleWalker {
+  // ** RULE IMPLEMENTATION HERE **
+  private spaced: boolean;
+  private exceptionsArrayOptions = [];
+  private braceException: boolean;
+  private bracketException: boolean;
+  private parenException: boolean;
+  private empty: boolean;
+
+  constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
+    super(sourceFile, options);
+    let ruleOptions = this.getOptions();
+    this.spaced = this.hasOption(ALWAYS) || ( ruleOptions && ruleOptions.length === 0);
+
+    if ( ruleOptions[1] ) {
+      this.exceptionsArrayOptions = (ruleOptions.length === 2) ? ruleOptions[1].exceptions : [] ;
+      if (this.exceptionsArrayOptions.length) {
+        this.braceException = this.exceptionsArrayOptions.indexOf('{}') !== -1;
+        this.bracketException = this.exceptionsArrayOptions.indexOf('[]') !== -1;
+        this.parenException = this.exceptionsArrayOptions.indexOf('()') !== -1;
+        this.empty = this.exceptionsArrayOptions.indexOf('empty') !== -1;
+      }
+    }
+  }
+
+  private getExceptions() {
+    let openers = [];
```

Comment:
These can be const since you are not changing reassigning. Would be handy to have http://eslint.org/docs/rules/prefer-const available.